rs/ic_os/config/src/lib.rs

```diff
@@ -31,8 +44,139 @@ fn parse_config_line(line: &str) -> Option<(String, String)> {
+                bail!("Invalid IPv6 prefix: {}", prefix);
+            }
+            format!("{}::", prefix)
+                .parse::<Ipv6Addr>()
+                .context(format!("Failed to parse IPv6 prefix: {}", prefix))
+        })
+        .transpose()?;
+
+    // Per PFOPS - ipv6_subnet will never not be 64
+    let ipv6_subnet = 64_u8;
+    // Optional ipv6_address - for testing. Takes precedence over ipv6_prefix.
+    let ipv6_address = config_map
+        .get("ipv6_address")
+        .map(|address| {
+            // ipv6_address might be formatted with the trailing suffix. Remove it.
+            address
+                .strip_suffix(&format!("/{}", ipv6_subnet))
+                .unwrap_or(address)
+                .parse::<Ipv6Addr>()
+                .context(format!("Invalid IPv6 address: {}", address))
+        })
+        .transpose()?;
+
+    if ipv6_address.is_none() && ipv6_prefix.is_none() {
+        bail!("Missing config parameter: need at least one of ipv6_prefix or ipv6_address");
+    }
+
+    let ipv6_gateway = config_map
+        .get("ipv6_gateway")
+        .context("Missing config parameter: ipv6_gateway")?
+        .parse::<Ipv6Addr>()
+        .context("Invalid IPv6 gateway address")?;
+
+    let ipv4_address = config_map
+        .get("ipv4_address")
+        .map(|address| {
+            address
+                .parse::<Ipv4Addr>()
+                .context(format!("Invalid IPv4 address: {}", address))
+        })
+        .transpose()?;
+
+    let ipv4_gateway = config_map
+        .get("ipv4_gateway")
+        .map(|address| {
+            address
+                .parse::<Ipv4Addr>()
+                .context(format!("Invalid IPv4 gateway: {}", address))
+        })
+        .transpose()?;
+
+    let ipv4_prefix_length = config_map
+        .get("ipv4_prefix_length")
+        .map(|prefix| {
+            let prefix = prefix
+                .parse::<u8>()
+                .context(format!("Invalid IPv4 prefix length: {}", prefix))?;
+            if prefix > 32 {
+                bail!(
+                    "IPv4 prefix length must be between 0 and 32, got {}",
+                    prefix
+                );
+            }
+            Ok(prefix)
+        })
+        .transpose()?;
+
+    let domain = config_map.get("domain").cloned();
+
+    let networking = crate::types::NetworkSettings {
+        ipv6_prefix,
+        ipv6_address,
+        ipv6_gateway,
+        ipv4_address,
+        ipv4_gateway,
+        ipv4_prefix_length,
+        domain,
+    };
+
+    let verbose = config_map
+        .get("verbose")
+        .map(|s| s.eq_ignore_ascii_case("true"))
+        .unwrap_or(false);
+
+    Ok((networking, verbose))
+}
+
+pub fn serialize_and_write_config<T: Serialize>(path: &Path, config: &T) -> Result<()> {
+    let serialized_config =
+        serde_json::to_string_pretty(config).expect("Failed to serialize configuration");
+    write_to_file(path, &serialized_config)
+}
```

Comment:
Are we in agreement about wanting to serialize to JSON?

Reply:
This is fine for this PR. No one is reading the serialized struct. We can decide on this later.